veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,258 @@
+# VEP #21: Seamless TCP migration and vhost-user support with passt
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue created, linking to VEP directory in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+KubeVirt users can benefit from seamless live migration of VMs while preserving TCP connections.
+This functionality has recently been introduced in `passt` together with `vhost-user` support, which increases network throughput. 
+This proposal outlines the integration plan to be implemented in KubeVirt. 
+
+## Motivation
+
+Currently, KubeVirt does not support all Primary Network requirements expected by users:
+- Seamless network connectivity on migration.
+- Observability tools such as [Stackrox Collector](https://github.com/stackrox/collector).
+- Service Mesh technologies such as [Istio](https://istio.io) and [Linkerd](https://linkerd.io/).
+- High throughput (on par with tap)
+
+Bridge binding does not support observability or service meshes, Masquerade hides the real IP from the user, and passt until recently lacked seamless migration and tap-level performance.
+These gaps have been recently addressed in passt r[release 2025_02_17.a1e48a0](https://passt.top/passt/tag/?h=2025_02_17.a1e48a0).
+
+## Future potential enhancement
+
+Support for other IP and non-IP network protocols, in addition to the currently supported TCP/UDP/ICMP.
+This capability is planned to be implemented in future releases of passt, and it too is expected to require adaptation in KubeVirt.
+
+## Goals
+
+- Seamless live migration of TCP connections without disconnection for Primary Network.
+- Support for observability tools and service mesh technologies.
+- Enhanced IPv6 support.
+- High network throughput, on par with Masquerade and Bridge bindings.
+
+## Non Goals
+
+- Supporting network protocols other those that are [currently supported](https://passt.top/passt/about/#protocols).
+- Supporting CNIs that do not provide sticky IP addresses during migration (migration succeeds, but existing connections break).
+
+## Definition of Users
+
+- Cluster Admins
+- VM/Namespace owners
+- Guest VM users – typically unaware of their network binding
+
+## User Stories
+
+- As a Guest VM user, I want optimal network performance comparable to bridge and masquerade, ensuring my virtualized applications perform well.  
+- As a Cluster Admin and as a Namespace Owner, I want observability and service meshes for enhanced monitoring, encryption, and load balancing.
+- As a Cluster Admin and as a Namespace Owner, I require seamless VM migration to perform infrastructure upgrades without affecting business continuity, preserving connectivity event for sensitive applications and protocols.
```

Comment:
Shouldn't it be the first user story?
The whole solution was built around this requirement.